src/index.ts

```diff
@@ -296,18 +298,57 @@ class Dataset<T extends Fields> {
     });
   }
 
+  replaceDateObjects(items: DatasetDataItem<T>[]): DatasetDataItem<T>[] {
+    const dateFields: Array<keyof T> = [];
+    const dateTimeFields: Array<keyof T> = [];
+
+    const keys = Object.keys(this.fields);
+    keys.forEach((fieldName) => {
+      const field = this.fields[fieldName];
+      if (field.type === 'date') {
+        dateFields.push(fieldName);
+      }
+      if (field.type === 'datetime') {
+        dateTimeFields.push(fieldName);
+      }
+    });
+    return items.map((item) => {
+      dateFields.forEach((fieldName) => {
+        const fieldValue = item[fieldName as keyof DatasetDataItem<T>];
+        if (fieldValue instanceof Date) {
+          item = {
+            ...item,
+            [fieldName]: (fieldValue as Date).toISOString().split('T')[0],
```

Comment:
using `toISOString()` here will convert the date to UTC, which might not be what we want for `Date` fields without time, but I could go either way on this one

Reply:
I think it's better to convert to UTC in both date and datetime cases, for consistency. It also means using using `new Date('2023-12-01')` will work as expected, even if maybe it shouldn't. From MDN "When the time zone offset is absent, date-only forms are interpreted as a UTC time and date-time forms are interpreted as local time. This is due to a historical spec error that was not consistent with ISO 8601 but could not be changed due to web compatibility"

Reply:
fair enough, my suggested update to the docs is a bit vague now - we should probably be more explicit about exactly how dates are handled including examples

Reply:
Yes, I'll add some more specific examples (and tests!)